### PR TITLE
[SPARK-14341] [SQL] Throw exception on unsupported create / drop macro ddl

### PR DESCRIPTION
**Pull request by bomeng**
## What changes were proposed in this pull request?

We throw an AnalysisException that looks like this:

```
scala> sqlContext.sql("CREATE TEMPORARY MACRO SIGMOID (x DOUBLE) 1.0 / (1.0 + EXP(-x))")
org.apache.spark.sql.catalyst.parser.ParseException:
Unsupported SQL statement
== SQL ==
CREATE TEMPORARY MACRO SIGMOID (x DOUBLE) 1.0 / (1.0 + EXP(-x))
  at org.apache.spark.sql.catalyst.parser.AbstractSqlParser.nativeCommand(ParseDriver.scala:66)
  at org.apache.spark.sql.catalyst.parser.AbstractSqlParser$$anonfun$parsePlan$1.apply(ParseDriver.scala:56)
  at org.apache.spark.sql.catalyst.parser.AbstractSqlParser$$anonfun$parsePlan$1.apply(ParseDriver.scala:53)
  at org.apache.spark.sql.catalyst.parser.AbstractSqlParser.parse(ParseDriver.scala:86)
  at org.apache.spark.sql.catalyst.parser.AbstractSqlParser.parsePlan(ParseDriver.scala:53)
  at org.apache.spark.sql.SQLContext.parseSql(SQLContext.scala:198)
  at org.apache.spark.sql.SQLContext.sql(SQLContext.scala:749)
  ... 48 elided

```
## How was this patch tested?

Add test cases in HiveQuerySuite.scala

### CODE REVIEW

**Comment by bomeng** [comment]
@hvanhovell could please help me on this? I thought this change was straightforward, but how come it failed org.apache.spark.sql.execution.streaming.state.StateStoreSuite twice? Also I can run the tests without errors on my local machine. Could not figure out. Thanks in advance.

**Comment by gatorsmile** [comment]
retest this please

**Comment by hvanhovell** [comment]
test this please

**Comment by dilipbiswal** [comment]
@hvanhovell @bomeng I see that we hit the same error for other PRs as well. - FYI

File: sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4 [hunk 1]
```diff
@@ -759,6 +761,7 @@ SNAPSHOT: 'SNAPSHOT';
 READ: 'READ';
 WRITE: 'WRITE';
 ONLY: 'ONLY';
+MACRO: 'MACRO';
```
Comment (hvanhovell) [inline sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: Could you add this keyword to the nonReserved rule?

Comment (bomeng) [inline sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: Just wanna confirm. I thought MACRO is a reseved keyword. Is it right? https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL

Comment (hvanhovell) [inline sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: Yeah you are right.

**Comment by hvanhovell** [comment]
LGTM

**Comment by hvanhovell** [comment]
Merging to master. Thanks!

**Pull request closed without merging**